api/app/serializers/serializer_registry.rb

```diff
@@ -53,7 +53,7 @@ def typed_attribute(attribute, type, options = {}, &block)
   end
 
   def abilities
-    typed_attribute :abilities, ::Types::Hash do |object, params|
+    typed_attribute :abilities, ::Types::Hash.meta(read_only: true) do |object, params|
```

Comment:
The attributes are always read_only, correct?

Reply:
You mean the abilities? Yes, abilities are always read only.